flyteplugins/go/tasks/plugins/webapi/agent/config.go

```diff
@@ -40,10 +40,12 @@ var (
 			},
 		},
 		DefaultAgent: Agent{
-			Endpoint:       "dns:///flyteagent.flyte.svc.cluster.local:80",
+			Endpoint:       "",
 			Insecure:       true,
 			DefaultTimeout: config.Duration{Duration: 10 * time.Second},
 		},
+		// AsyncPlugin should be registered to at least one task type.
+		// Reference: https://github.com/flyteorg/flyte/blob/master/flyteplugins/go/tasks/pluginmachinery/registry.go#L27
 		SupportedTaskTypes: []string{"task_type_1", "task_type_2"},
```

Comment:
Maybe we should just remove `SupportedTaskTypes` to ship the breaking change and from now on we will solely rely on metadata as the single source of truth. If we still want to support this config, we will definitely need to be more resilient when applying the routing to find agent deployment because users might have a bad configuration, more specifically I am referring to `getFinalAgent` function.